setup.py

```diff
@@ -26,6 +26,12 @@
     package_data.setdefault('setuptools.command', []).extend(['*.xml'])
 
 
+def get_version() -> str:
+    version_files = (f for f in [".latest", ".stable"] if os.path.exists(f))
+    with open(next(version_files), encoding="utf-8") as fp:
+        return fp.read()
+
+
```

Comment:
The way I see, the `setup.py` file is an inevitable part of the ecosystem, and probably it is here to stay (not as a CLI tool, but as a Python configuration file like many others, e.g. `conftest.py`, `noxfile.py`, etc...).

If not `setup.py` then we would end up with something very similar and "imperative"/"programatic". For example, `hatch` docs suggest a [hatch_build.py](https://hatch.pypa.io/1.12/plugins/builder/custom/#hatch_buildpy) script as means of configuration. Granted it uses a different API and requires a class definition rather than calling a `setup()` function, but at that point, I don't think it is worthy re-inventing a new API, since `setuptools.setup()` is already well known.

I think `setup.py` files are useful for 2 reasons:

- It is very practical for users to implement customisations as part of the local directory.
- Avoid the need for defining a full-blown plugin when the customisation is particular to a single project and the logic does not generalise well.


In this case, I think that the logic "get the content from the first of 2 files to exist" sounds very particular to this implementation, therefore `setup.py` sounds like the proper place to write it.